backend/danswer/connectors/teams/connector.py

```diff
@@ -0,0 +1,281 @@
+import os
+from datetime import datetime
+from datetime import timezone
+from typing import Any
+
+import msal  # type: ignore
+from office365.graph_client import GraphClient  # type: ignore
+from office365.teams.channels.channel import Channel  # type: ignore
+from office365.teams.chats.messages.message import ChatMessage  # type: ignore
+from office365.teams.team import Team  # type: ignore
+
+from danswer.configs.app_configs import INDEX_BATCH_SIZE
+from danswer.configs.constants import DocumentSource
+from danswer.connectors.interfaces import GenerateDocumentsOutput
+from danswer.connectors.interfaces import LoadConnector
+from danswer.connectors.interfaces import PollConnector
+from danswer.connectors.interfaces import SecondsSinceUnixEpoch
+from danswer.connectors.models import BasicExpertInfo
+from danswer.connectors.models import ConnectorMissingCredentialError
+from danswer.connectors.models import Document
+from danswer.connectors.models import Section
+from danswer.file_processing.html_utils import parse_html_page_basic
+from danswer.utils.logger import setup_logger
+
+logger = setup_logger()
+
+datetime_format_string = "%Y-%m-%dT%H:%M:%S.%f%z"
+
+
+def get_created_datetime(chat_message: ChatMessage) -> datetime:
```

Comment:
nit: this could just reuse the existing function. Also let's be sure that all Datetime objects have timezone info where possible.